### PR TITLE
Unit tests for search()

### PR DESCRIPTION
**Pull request by fejta**
/assign @cjwagner @stevekuznetsov 

Validate that we:
* Return partial results on failure
* Include correct start/end time in query
* Page correctly

#11679

### CODE REVIEW

File: prow/tide/search_test.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	githubql "github.com/shurcooL/githubv4"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/util/diff"
+)
+
+func TestSearch(t *testing.T) {
+	const q = "random search string"
+	now := time.Now()
+	earlier := now.Add(5 * time.Hour)
```
Comment (cjwagner) [inline prow/tide/search_test.go]: This is later than `now` not earlier.

File: prow/tide/search_test.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+			q:       datedQuery(q, earlier, now),
+			cursors: []*githubql.String{nil},
+			sqs: []searchQuery{
+				{},
+			},
+			errs: []error{errors.New("injected error")},
+			err:  true,
+		},
+		{
+			name:    "set minimum start time",
```
Comment (cjwagner) [inline prow/tide/search_test.go]: I don't think this actually validates that the start time is floored as expected.

Comment (fejta) [reply prow/tide/search_test.go]: It does. We set the start time to `time.Time{}` (aka zero time). We validate this gets floored to `floor(time.Time{})` in the query

File: prow/tide/search_test.go (hunk 1 above)
Comment (fejta) [reply prow/tide/search_test.go]: Done

File: prow/tide/search_test.go (hunk 2 above)
Comment (cjwagner) [reply prow/tide/search_test.go]: Ah I see. This doesn't test that the behavior of `floor` is correct, it just tries to test that `floor` is called assuming the behavior is correct (which is reasonable given its simplicity).

**Pull request merged**